os/arch/arm/src/amebalite/amebalite_serial.c

```diff
@@ -818,8 +818,15 @@ static int rtl8720e_up_setup(struct uart_dev_s *dev)
 	serial_init((serial_t *) sdrv[uart_index_get(priv->tx)], priv->tx, priv->rx);
 	serial_baud(sdrv[uart_index_get(priv->tx)], priv->baud);
 	serial_format(sdrv[uart_index_get(priv->tx)], priv->bits, priv->parity, priv->stopbit);
-	serial_set_flow_control(sdrv[uart_index_get(priv->tx)], priv->FlowControl, priv->rts, priv->cts);
+
+#if defined(CONFIG_SERIAL_IFLOWCONTROL) || defined(CONFIG_SERIAL_OFLOWCONTROL)
+	if ((sdrv[uart_index_get(priv->tx)] == 0) || (sdrv[uart_index_get(priv->tx)] == 3)) {
+		serial_set_flow_control(sdrv[uart_index_get(priv->tx)], priv->FlowControl, priv->rts, priv->cts);
+	} else {
+		printf("\nSetting UART%d flow control failed.\n", sdrv[uart_index_get(priv->tx));
```

Comment:
It looks that this function is related to uart configuration.
Is it okay to use printf in this function?

Reply:
Yes, this printf is used to output at loguart side

Reply:
@jeongchanKim means this function is called to setup UART but this printf is called with failure. How do you use printf with uart setup failure? The printf uses uart.

Reply:
The printf is tied to loguart uart4 which will be init first and printf will head towards there
, which is in separate function (else condition) from UART0/1/2/3 for flow control
!
[Capture](https://user-images.githubusercontent.com/61275451/186621522-2aa0fa50-6889-4776-840f-b7a3a5b8f860.PNG)
l